### PR TITLE
ipmitool: add enterprise-numbers file

### PR DESCRIPTION
**Pull request by posch**
###### Description of changes

Current ipmitool complains about missing enterprise-numbers file. See issue #204926
https://github.com/NixOS/nixpkgs/issues/204926

ipmitool tries to download the file from iana.org:
https://www.iana.org/assignments/enterprise-numbers.txt

It is quite large (~4.5 MB, ~1.5 MB compressed), so we may not want to include it in nixpkgs directly. I've set up a github repository that contains the file from iana.org, and changed default.nix to fetch it.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickCao** [comment]
Or we can use the internet archive.

**Comment by posch** [comment]
> Or we can use the internet archive.

Indeed, I hadn't thought of this, and nixpkgs is already using archive.org quite a number of times.

There was only one version of enterprise-numbers.txt in the wayback machine. I used "Save Page Now" to create an up-to-date copy and modified default.nix to use it.

File: pkgs/tools/system/ipmitool/default.nix [hunk 1]
```diff
@@ -16,7 +23,9 @@ stdenv.mkDerivation rec {
 
   postPatch = ''
     substituteInPlace configure.ac \
-      --replace 'AC_MSG_WARN([** Neither wget nor curl could be found.])' 'AM_CONDITIONAL([DOWNLOAD], [false])'
+      --replace 'AC_MSG_WARN([** Neither wget nor curl could be found.])' 'AM_CONDITIONAL([DOWNLOAD], [true])'
+
+    cat ${iana-enterprise-numbers} >enterprise-numbers
```
Comment (NickCao) [inline pkgs/tools/system/ipmitool/default.nix:28]: Just use cp or ln here.

Comment (posch) [reply pkgs/tools/system/ipmitool/default.nix]: Sure. ln doesn't work (Invalid cross-device link). cp does.

**Comment by NickCao** [comment]
And lastly, squash the commits. diff LGTM, just need to wait for ofborg to test it out.

**Review by NickCao — APPROVED** [review]
checked build output, the enterprise-numbers file is installed into the correct place.

**Comment by melvyn2** [comment]
will this be backported to 22.11?

**Pull request merged**